python/pylibcudf/pylibcudf/aggregation.pyx

```diff
@@ -193,6 +195,17 @@ cdef class Aggregation:
             self._unsupported_agg_error("rolling")
         return agg_cast
 
+    cdef const unique_ptr[rolling_aggregation] clone_underlying_as_rolling(
```

Comment:
Can we move this impl up with the other `clone*` methods?

It's sort of an implementation detail that some of these require cloning while others require viewing. Maybe it would make sense to have a single clone method that produces a raw aggregation, then you could call the suitable view* method on that? That might not be possible within the limitations of Cython's ability to handle smart pointer casting though.

Reply:
Move up, but didn't try the other thing,